monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/IntersperseObservable.scala

```diff
@@ -0,0 +1,56 @@
+package monix.reactive.internal.operators
+
+import monix.execution.Ack.Continue
+import monix.execution.atomic.AtomicBoolean
+import monix.execution.{Ack, Cancelable}
+import monix.reactive.Observable
+import monix.reactive.observers.Subscriber
+
+import scala.concurrent.Future
+
+/**
+  * Created by omainegra on 5/29/17.
+  */
+private[reactive] final class IntersperseObservable[+A](source: Observable[A],
+                                                        separator: A) extends Observable[A]{ self =>
+
+  override def unsafeSubscribeFn(out: Subscriber[A]): Cancelable = {
+    val firstTime = AtomicBoolean(true)
```

Comment:
You are modifying `firstTime` in `onNext` only, so you don't need to synchronize it, because the `onNext` calls are guaranteed to be ordered (the same guarantee you get with actors processing messages).

Atomics come with overhead and this can be a simple `private[this] var` in that `Subscriber` class.